docs/src/userguide/source_files_and_compilation.rst

```diff
@@ -833,6 +833,16 @@ Cython code.  Here is the list of currently supported directives:
     appropriate exception is raised. This is off by default for
     performance reasons.  Default is False.
 
+``freethreading_compatible``  (True / False)
```

Comment:
From another PR, it seems that it would be helpful to know whether the module targets freethreading Python or not, in order to take performance relevant code generation decisions. Is there an argument for reserving this option to "is compatible" rather than "wants to support"? If not, then I'd just call it `freethreading` and give it both meanings when enabled: "I accept slightly less optimal code generation because I want the module to be fully working without the GIL in a free-threading Python".

Reply:
I think the "do we want to generate slightly less efficient code" is probably a separate question .

The code generation seems like something you should be able to turn on and off locally based on your knowledge of how your own module will be used.

I think it'd be reasonable for this option to affect the default value for the code generation, since if a module isn't designed to be free-threaded then there's no need to slow things down to support it.

But after that, I think they should be separate directives.

Reply:
Agree with @da-woods here FWIW.